src/Services/FHIR/FhirEncounterService.php

```diff
@@ -96,13 +96,13 @@ public function parseOpenEMRRecord($dataRecord = array(), $encode = false)
             $encounterResource->setClass($class);
         }
 
-        // Encounter.type
-        $type = new FHIRCodeableConcept();
-        $type->addCoding(array(
-            "system" => "http://snomed.info/sct",
-            "code" => "185349003"
-        ));
-        $type->setText("Encounter for check up (procedure)");
+        if (!empty($dataRecord['pc_catid'])) {
+                $type = new FHIRCodeableConcept();
+                $encounterType = $this->getEncounterType($dataRecord['pc_catid']);
```

Comment:
We basically hard-coded the encounter type here to ensure it was compliant (note the list at the bottom of link):
http://hl7.org/fhir/us/core/ValueSet-us-core-encounter-type.html
(the codes in pc_catid are not on this valueset listing)

Reply:
One possible solution for this, adding a new field in the table form_encounter with column encounter_type

The value for the above mentioned column **encounter_type** can be populated from a master table name **encounter_types** with the values available in http://hl7.org/fhir/us/core/ValueSet-us-core-encounter-type.html

This will also lead to a change in UI, where the user will have to optionally select the encounter type. This selection can be similar to what is being used for find_code.  

If this solution is acceptable, I can go ahead and make the changes. 

Please let me know.

Reply:
hi @vishnuyar ,
Would prob consider following:
1. Add the items as a list in list_options (check out the us-core-provider-specialty listing in database.sql and 5_0_2-to-6_0_0_upgrade.sql to see how to add a list).
2. Then make this a choice in the Administration->Clinic->Calendar->Categories gui (this means that will define what kind of encounter type that choice is since this is what is selected in the 'Visit Category:' item when select a new encounter)(thus when a user selects a standard 'Visit Category' in the new encounter, it will be mapped to the chosen encounter type for that visit category). This avoids needing the user to have to select a encounter type every time they open new encounter. To see how to create a selector from a list in list_options, check out use of the generate_select_list() function in the codebase.

Reply:
I have undone the changes made to FHIREncounterService.php. It sets the encounter type to currently hardcoded one.  Subsequently I will work on encounter types based on the suggested solution

Reply:
Added another quick fix to the current pull request.  In FHIR Encounter resource the Practitioner reference value should uuid instead of provider_id, With this change, we can get the details of the FHIR practitioner resource using the uuid from Enounter FHIR Resource.